Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/CclJsonFunctions.kt

```diff
@@ -0,0 +1,53 @@
+package de.rki.coronawarnapp.ccl.dccwalletinfo.calculation
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import de.rki.coronawarnapp.ccl.configuration.model.CCLConfiguration
+import de.rki.coronawarnapp.ccl.configuration.storage.CCLConfigurationRepository
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.serialization.BaseJackson
+import de.rki.jfn.JsonFunctions
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CclJsonFunctions @Inject constructor(
+    @BaseJackson private val mapper: ObjectMapper,
+    @AppScope private val appScope: CoroutineScope,
+    private val configurationRepository: CCLConfigurationRepository,
+) {
+
+    private lateinit var jsonFunctions: JsonFunctions
+
+    init {
+        appScope.launch {
+            jsonFunctions = create(configurationRepository.getCCLConfigurations())
+        }
+    }
+
+    fun update(cclConfigurations: List<CCLConfiguration>) {
+        jsonFunctions = create(cclConfigurations)
+    }
+
+    fun evaluateFunction(
+        functionName: String,
+        parameters: JsonNode
+    ) = jsonFunctions.evaluateFunction(functionName, parameters)
```

Comment:
No mutex necessary here?

Reply:
`evaluateFunction` itself has a mutex it is supposed to be thread save

Reply:
I think we need one here and another one in the `create()` function. Otherwise, if `evaluateFunction()` and `update()` are called from different threads, something could tried to be evaluated although not all functions have been registered yet.

Reply:
mutex added